contracts/DependencyAudit.cdc

```diff
@@ -41,27 +41,99 @@ access(all) contract DependencyAudit {
         }
 
         if unstagedDependencies.length > 0 {
-            if DependencyAudit.panicOnUnstaged {
-                // If `panicOnUnstaged` is set to true, the transaction will panic if there are any unstaged dependencies
-                // the panic message will include the unstaged dependencies
-                var unstagedDependenciesString = ""
-                var numUnstagedDependencies = unstagedDependencies.length
-                var j = 0
-                while j < numUnstagedDependencies {
-                    if j > 0 {
-                        unstagedDependenciesString = unstagedDependenciesString.concat(", ")
-                    }
-                    unstagedDependenciesString = unstagedDependenciesString.concat(unstagedDependencies[j].toString())
-
-                    j = j + 1
-                }
-
-                // the transactions will fail with a message that looks like this: `error: panic: Found unstaged dependencies: A.2ceae959ed1a7e7a.MigrationContractStaging, A.2ceae959ed1a7e7a.DependencyAudit`
-                panic("This transaction is using dependencies not staged for Crescendo upgrade coming soon! Learn more: https://bit.ly/FLOWCRESCENDO. Dependencies not staged: ".concat(unstagedDependenciesString))
-            } else {
-                emit UnstagedDependencies(dependencies: unstagedDependencies)
+            self.maybePanicOnUnstagedDependencies(unstagedDependencies)
+
+            emit UnstagedDependencies(dependencies: unstagedDependencies)
+        }
+    }
+
+    access(self) fun maybePanicOnUnstagedDependencies(_ unstagedDependencies: [Dependency]) {
+        // If `panicOnUnstaged` is set to false, the function will return without panicking
+        if !DependencyAudit.panicOnUnstaged {
+            return
+        }
+
+        // check if we should panic randomly
+        if !self.shouldPanicRandomly() {
```

Comment:
Could probably consolidate this check as a disjunction with the previous check, short circuiting on `panicOnUnstaged`

Reply:
Is this what you had in mind? 
636670d